src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,108 @@
+import { type ChangeEvent, useCallback, useEffect, useState } from 'react';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { maxSlippage, setMaxSlippage } = useSwapContext();
```

Comment:
Added the context changes because Input depends on it

Reply:
The "only" thing you need to import from the context is:
- lifeCycleStatus
- setLifeCycleStatus

So write code in a way that only these two can be used.

I am not going to write you directly here what you have to do. This is a System Design challenge that you need to learn to feel confortable to solve.

I am going to let you cook this on your own, and than if you don't get closer to hit, by Sunday, I will give you the answear.

My only hint is, look @abcrane123 style of coding in the recent PRs, and try to think what she did. If you understand what she did, you will be one step closer to your answear.

Reply:
Just saw this. I think my current iteration is close. Let me know what you think and still let me cook on it tomorrow if it's not there yet